packages/block-library/src/table/edit.js

```diff
@@ -291,14 +291,24 @@ export class TableEdit extends Component {
 
 		const { attributes, setAttributes } = this.props;
 		const { sectionName, rowIndex } = selectedCell;
+		const newRowIndex = rowIndex + delta;
 
-		this.setState( { selectedCell: null } );
 		setAttributes(
 			insertRow( attributes, {
 				sectionName,
-				rowIndex: rowIndex + delta,
+				rowIndex: newRowIndex,
 			} )
 		);
+		// this just allows to be able to add more rows, by enabling the corresponding
+		// buttons. It needs to be revised when programmatically focus to RichText is handled
```

Comment:
```suggestion
		// Select the first cell of the new row
```

I think the above is enough. Right now we have an adequate solution; if we had something more "hacky", some context would indeed be useful. And we already have the RichText issue open for any future enhancements.